tree/dataframe/test/dataframe_incomplete_entries.cxx

```diff
@@ -1093,16 +1093,116 @@ TEST_P(RDFMissingBranchesVec, DefaultValueForSimpleVariation)
    }
 }
 
+struct RDFIndexedFriendsNotOrdered : public ::testing::TestWithParam<bool> {
+   constexpr static std::array<const char *, 3> fFileNames{"dataframe_friends_chain_unordered_index_column_1.root",
+                                                           "dataframe_friends_chain_unordered_index_column_2.root",
+                                                           "dataframe_friends_chain_unordered_index_column_3.root"};
+   constexpr static auto fMainTreeName{"main_tree"};
+   constexpr static auto fAuxTreeName{"aux_tree"};
+   const unsigned int fNSlots;
+
+   RDFIndexedFriendsNotOrdered() : fNSlots(GetParam() ? std::min(4u, std::thread::hardware_concurrency()) : 1u)
+   {
+      if (GetParam())
+         ROOT::EnableImplicitMT(fNSlots);
+   }
+
+   ~RDFIndexedFriendsNotOrdered() override
+   {
+      if (GetParam())
+         ROOT::DisableImplicitMT();
+   }
+
+   static void SetUpTestSuite()
+   {
+      for (std::size_t i = 0; i < fFileNames.size(); i++) {
+         TFile f(fFileNames[i], "RECREATE");
+         TTree main_tree(fMainTreeName, fMainTreeName);
+         int evt_number;
+         int x;
+         main_tree.Branch("evt_number", &evt_number);
+         main_tree.Branch("x", &x);
+
+         for (auto j = 0; j < 5; j++) {
+            evt_number = 5 * i + j;
+            x = evt_number;
+            main_tree.Fill();
+         }
+
+         TTree aux_tree{fAuxTreeName, fAuxTreeName};
+         int evt_aux;
+         int y;
+         aux_tree.Branch("evt_number", &evt_aux);
+         aux_tree.Branch("y", &y);
+         evt_aux = 5 * i + 4;
+         y = 10 * (i + 1);
+         aux_tree.Fill();
+         evt_aux = 5 * i + 1;
+         y = 11 * (i + 1);
+         aux_tree.Fill();
+
+         f.Write();
+      }
+   }
+
+   static void TearDownTestSuite()
+   {
+      for (auto &&fileName : fFileNames)
+         std::remove(fileName);
+   }
+};
+
+TEST_P(RDFIndexedFriendsNotOrdered, SuppressIndexOrderingError)
```

Comment:
The test seems to be failing.  Is it known why?

Reply:
Not really, in fact it does not fail on all platforms. Needs to be investigated